### ISSUE TITLE
Problems with StarNode.js hinder TypeScript conversion.

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/scenery-phet#734]
These problems were encountered when attempting to convert StarNode.js to TypeScript.  They were serious enough that I bailed.  The changes are significant, and thorough testing will be required. 

(1) The first problem was introduced by @samreid in https://github.com/phetsims/scenery-phet/commit/1ce9b8d8:

```js
    const o2 = _.clone( options );
    o2.value = 1;
    const backgroundStarShape = new StarShape( o2 );
...
      const foregroundStar = new Path( new StarShape( o2 ), {
...
      foregroundStar.shape = new StarShape( o2 );
```

`StarShape` has no `value` option.  And the Shape for `foregroundStar` is unnecessarily created twice.

(2) The second problem was introduced by  @jbphet in https://github.com/phetsims/scenery-phet/commit/fe6f0d68:

```js
    function getBounds() {
      return backgroundStarShape.bounds;
    }
...
      backgroundStar.computeShapeBounds = getBounds; // optimization - override bounds calculation to used pre-computed value
...
      foregroundStar.computeShapeBounds = getBounds; // optimization - override bounds calculation to used pre-computed value
```

This the nuclear approach to overriding a method.  What should have been done is to create a subclass of Path (maybe call it `OptimizedStarPath`) that overrides `computeShapeBounds`.

It's also not clear if  `foregroundStar`should be based on `backgroundStarShape.bounds`, but that's what is happening here.

**Comment by pixelzoom** [comment]
While converting vegas code, I encountered the need for StarNodeOptions in several places.  For example in ScoreDisplayNumberAndStar.ts:

```js
type SelfOptions = {
  font?: Font;
  textFill?: IColor;
  scoreDecimalPlaces?: number;
  spacing?: number;
  starNodeOptions?: any; //TODO https://github.com/phetsims/scenery-phet/issues/734
};
```

I've added a TODOs referencing this issue, so please look for those.

**Comment by jbphet** [comment]
I've addressed the issues that @pixelzoom laid out above, converted `StarNode` to TypeScript, and changed the places where options were used to control this node.  I regression tested these changes in the vegas demo, the scenery-phet demo, the score displays in arithmetic, the reward node in expression-exchange, and the "target hit" graphic in projectile motion.  I think I got everything, but this is pretty widely used, so I could easily have missed something.  @pixelzoom - care to do a review?
